src/main/java/revir/user/Parser.java

```diff
@@ -0,0 +1,144 @@
+package revir.user;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.HashMap;
+import java.util.Map;
+
+import revir.system.exceptions.IllegalCommandException;
+import revir.system.exceptions.InvalidFormatException;
+import revir.tasks.Deadline;
+import revir.tasks.Event;
+import revir.tasks.Todo;
+import revir.user.command.Command;
+import revir.user.command.Create;
+import revir.user.command.Delete;
+import revir.user.command.Find;
+import revir.user.command.ListTasks;
+import revir.user.command.Mark;
+import revir.user.command.Nop;
+
+/**
+ * Parses the command string and returns the corresponding Command object.
+ */
+public class Parser {
+    private enum Action {
+        BYE, LIST, MARK, UNMARK, TODO, DEADLINE, EVENT, DELETE, FIND, INVALID
+    }
+
+    private static DateTimeFormatter dateFormat = DateTimeFormatter.ofPattern("d/M/yyyy HHmm");
+    private static final Map<String, Action> actionMap = new HashMap<>();
+    static {
+        actionMap.put("bye", Action.BYE);
+        actionMap.put("list", Action.LIST);
+        actionMap.put("mark", Action.MARK);
+        actionMap.put("unmark", Action.UNMARK);
+        actionMap.put("todo", Action.TODO);
+        actionMap.put("deadline", Action.DEADLINE);
+        actionMap.put("event", Action.EVENT);
+        actionMap.put("delete", Action.DELETE);
+        actionMap.put("find", Action.FIND);
+    }
+
+    /**
+     * Parses the command string and returns the corresponding Command object.
+     *
+     * @param commandString The command string to parse.
+     * @return The Command object corresponding to the command string.
+     * @throws IllegalCommandException If the command is not recognized.
+     * @throws InvalidFormatException If the command is recognized but the format is invalid.
+     */
+    public Command parse(String commandString) throws IllegalCommandException, InvalidFormatException {
+        String[] split = commandString.trim().split(" ", 2);
```

Comment:
'split' seems like a suitable name to me, but according to Java naming conventions, collections should be named in plural.